HaskellDates.hsproj/HaskellDates.hs

```diff
@@ -0,0 +1,62 @@
+import Data.Time
+import Data.Time.Calendar.OrdinalDate
+import Control.Applicative
+
+data Event = Event {
+  date :: Day,
+  daysOfWeek :: [Int],
+  week :: Int
+}
+
+-- WeekDay for debugging purposes
+newtype WeekDay = WeekDay Int
+
+instance Show WeekDay where
+  show (WeekDay x)
+    | x == 0 = "Sunday"
+    | x == 1 = "Monday"
+    | x == 2 = "Tuesday"
+    | x == 3 = "Wednesday"
+    | x == 4 = "Thursday"
+    | x == 5 = "Friday"
+    | x == 6 = "Saturday"
+
+weekDay :: Day -> WeekDay
+weekDay = WeekDay . dayOfWeek
+
+-- return the day of the week
+dayOfWeek :: Day -> Int
+dayOfWeek = snd . sundayStartWeek
+
+-- return the month of the day
+toMonth :: Day -> Int
+toMonth = middle . toGregorian
+  where middle (_, x, _) = x
+
+-- for example: 1 = first day occurance in month (first Wednesday)
+-- 2 = second occurance (second Wednesday)
+weekDayOfMonth :: Day -> Int
+weekDayOfMonth day
+  | (toMonth lastWeek) /= month = 1
```

Comment:
I tried without but it was still giving me errors ... I'll check again to be sure